loopy/check.py

```diff
@@ -445,19 +464,11 @@ def map_subscript(self, expr, domain):
                         % (expr, self.insn_id, access_range, shape_domain))
 
     def map_if(self, expr, domain):
-        from loopy.symbolic import get_dependencies
-        if get_dependencies(expr.condition) <= frozenset(
-                domain.space.get_var_dict()):
-            try:
-                from loopy.symbolic import isl_set_from_expr
-                then_set = isl_set_from_expr(domain.space, expr.condition)
-                else_set = then_set.complement()
-            except ExpressionToAffineConversionError:
-                # non-affine condition: can't do much
-                then_set = else_set = isl.BasicSet.universe(domain.space)
-        else:
-            # data-dependent condition: can't do much
+        then_set = _condition_to_set(domain.space, expr.condition)
+        if then_set is None:
             then_set = else_set = isl.BasicSet.universe(domain.space)
```

Comment:
That's because our inference of expression to an ISL set failed, in that case we check  both the branches of the "if" without any additional assumptions from the condition expression.

Reply:
OIC, fair. Could you add a comment to explain this?

Reply:
Sure! 7421187